diesel/src/query_builder/combination_clause.rs

```diff
@@ -32,41 +56,88 @@ impl<Combinator, Rule, Source, Rhs> CombinationClause<Combinator, Rule, Source,
             duplicate_rule,
             source: ParenthesisWrapper(source),
             rhs: ParenthesisWrapper(rhs),
+            limit_offset: LimitOffsetClause {
+                limit_clause: NoLimitClause,
+                offset_clause: NoOffsetClause,
+            },
         }
     }
 }
 
-impl<Combinator, Rule, Source, Rhs> Query for CombinationClause<Combinator, Rule, Source, Rhs>
+impl<Combinator, Rule, Source, Rhs, LOf>
+    CombinationClause<Combinator, Rule, Source, Rhs, LOf>
+{
+    pub(crate) fn new_full(
```

Comment:
I could remove this method and change `new` instead, but then I would need to propagate the LimitOffsetClause declaration to each call site, which happens in several files, so I'm not entirely sure about that and I'm looking for feedback on it.

The method name is subpar :/

Reply:
Given that we only use this function inside of this file I just wouldn't add it at all. Instead use the direct struct construction syntax at the calling sides:

```rust
 CombinationClause{
            combinator: self.combinator,
            duplicate_rule: self.duplicate_rule,
            source: self.source.0,
            rhs: self.rhs.0,
            limit_offest: LimitOffsetClause {
                limit_clause: self.limit_offset.limit_clause,
                offset_clause,
            },
        }
```

(I think that even looks clearer)